internal/wasm/sys_test.go

```diff
@@ -147,3 +150,57 @@ func TestNewSysContext_Environ(t *testing.T) {
 		})
 	}
 }
+
+func TestSysContext_Close(t *testing.T) {
+	t.Run("no files", func(t *testing.T) {
+		sys := DefaultSysContext()
+		require.NoError(t, sys.Close())
+	})
+
+	t.Run("open files", func(t *testing.T) {
+		tempDir := t.TempDir()
+		pathName := "test"
+		file, _ := createWriteableFile(t, tempDir, pathName, make([]byte, 0))
+
+		sys, err := NewSysContext(
+			0,   // max
+			nil, // args
+			nil, // environ
+			nil, // stdin
+			nil, // stdout
+			nil, // stderr
+			map[uint32]*FileEntry{ // openedFiles
+				3: {Path: "."},
+				4: {Path: path.Join(".", pathName), File: file},
+			},
+		)
+		require.NoError(t, err)
+
+		// Closing should delete the file descriptors after closing the files.
+		require.NoError(t, sys.Close())
+		require.Empty(t, sys.openedFiles)
+
+		// Verify it was actually closed, by trying to close it again.
+		err = file.Close()
+		require.Contains(t, err.Error(), "file already closed")
+
+		// No problem closing config again because the descriptors were removed, so they won't be called again.
+		require.NoError(t, sys.Close())
+	})
+
+	// TODO: fs but never used (ex file == nil)
```

Comment:
I won't be addressing these TODOs in this PR. This is just to unlock the other one. I'll do it in #394